test/print_test_stats.py

```diff
@@ -135,9 +135,21 @@ def send_report_to_scribe(reports):
 def send_report_to_s3(reports, *, total_seconds):
     job = os.environ.get('CIRCLE_JOB')
     sha1 = os.environ.get('CIRCLE_SHA1')
+    branch = os.environ.get('CIRCLE_BRANCH')
+    if branch != 'master':
+        print(f"S3 upload only enabled on master. skipping branch {branch}")
+        return
     now = datetime.datetime.utcnow().isoformat()
     key = f'test_time/{sha1}/{job}/{now}Z.json.bz2'  # Z meaning UTC
-    obj = boto3.resource('s3').Object('ossci-metrics', key)
+    s3 = boto3.resource('s3')
+    try:
+        s3.get_bucket_acl(Bucket='ossci-metrics')
+    except Exception as e:
+        print(f"AWS ACL failed: {e}")
```

Comment:
Why do we need to check for that?

Reply:
I added this because scribe upload also has credential check. i figure this would give us more debuggable info?

Reply:
note from a conversation @walterddr and I had just now: this seems to not be working as expected, since [the Circle logs](https://app.circleci.com/pipelines/github/pytorch/pytorch/256864/workflows/e4953b8d-db32-4292-afe2-2c3c65fd1e88/jobs/9961274/parallel-runs/0/steps/0-108) show that this is printing an error message and yet continuing to [successfully upload](https://s3.console.aws.amazon.com/s3/buckets/ossci-metrics?prefix=test_time/e4d596c575018eddab9d9b6be0e42319de51224b/):
```
Jan 05 17:12:28 AWS ACL failed: 's3.ServiceResource' object has no attribute 'get_bucket_acl'
Jan 05 17:12:28 AWS credential found, uploading to S3...
```
a couple of other notes from the same discussion:

> that could've been a typo. should be `s3.Client.get_bucket_acl(...)`

> if credential is missing, we shouldn't try uploading. but I guess alternative way to handle this would be: wrap around the entire upload block with try-catch, and printout exception when boto3 ACL fails

> e.g. we dont need to explicitly do the credential check. it will fail the upload anyway